libcloud/compute/drivers/azure_arm.py

```diff
@@ -912,6 +917,19 @@ def destroy_node(
                         raise
                 time.sleep(10)
 
+        # Optionally destroy the OS managed disk
+        managed_disk = node.extra["properties"]["storageProfile"]["osDisk"].get("managedDisk")
```

Comment:
Can we be sure that all those nested objects / dictionaries are indeed always present? Or it would be safer to use `.get("key", {})` notation for all the nested attributes?

Reply:
You're right. I didn't find anywhere on Azure documentation the properties are always there so I fell back to your proposal to make it safer. Thanks.